src/TypeCaster/HydratorTypeCaster.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Hydrator\TypeCaster;
+
+use ReflectionClass;
+use ReflectionNamedType;
+use ReflectionType;
+use ReflectionUnionType;
+use Yiisoft\Hydrator\HydratorInterface;
+use Yiisoft\Hydrator\Result;
+use Yiisoft\Hydrator\TypeCasterInterface;
+
+use function is_array;
+
+/**
+ * Cast arrays to objects.
+ */
+final class HydratorTypeCaster implements TypeCasterInterface
```

Comment:
I think name could be a bit more specific.

Reply:
Do you have any suggestions?

Reply:
What about `ListTypeCaster` or `CollectionTypeCaster`?